### PR TITLE
WASM Refactor

### PR DESCRIPTION
**Pull request by ziegfried**
- Switched to ethabi lib for ABI decoding (via Rust/WASM)
- Improved anonymous ABI decoding (collision handling, fingerprinting improvements)

### CODE REVIEW

File: wasm/ethabi/src/sig.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            match parse_fn_args(&String::from(args), allow_indexed_flag) {
+                Ok(args) => Ok(AbiItem {
+                    name: String::from(name),
+                    inputs: args,
+                }),
+                Err(msg) => Err(format!("{}", msg)),
+            }
+        }
+        None => Err(String::from(
+            "Unable to parse signature: no open parentesis found",
```
Comment (dexa187) [inline wasm/ethabi/src/sig.rs]:
```suggestion
            "Unable to parse signature: no open parenthesis found",
```

File: scripts/publish-ci-image.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable no-console */
+import execa from 'execa';
+import { join } from 'path';
+import { readFile, writeFile } from 'fs-extra';
+
+const IMAGE = 'splunkdlt/scfe-ci';
```
Comment (dexa187) [inline scripts/publish-ci-image.ts]: What does scfe stand for?

Comment (ziegfried) [reply scripts/publish-ci-image.ts]: Splunk Connect for Ethereum

Comment (dexa187) [reply scripts/publish-ci-image.ts]: ha cant believe i didnt get that

**Pull request merged**